lib/synapse/config_generator/haproxy.rb

```diff
@@ -1102,20 +1105,23 @@ def update_backends_at(socket_file_path, watchers)
 
       # parse the stats output to get current backends
       cur_backends = {}
+      re = Regexp.new('^(.+?),(.+?),(?:.*?,){15}(.+?),')
```

Comment:
This regex scares me a bit ... but ... ok

What's wrong with just splitting on ',' again?